Fortran/gfortran/regression/DisabledFiles.cmake

```diff
@@ -661,11 +588,6 @@ file(GLOB SKIPPED_FILES CONFIGURE_DEPENDS
   cray_pointers_7.f90
   dec_math.f90
   dec_math_5.f90
-  fmt_en.f90
-  fmt_en_rd.f90
```

Comment:
I am getting link errors here:
```
/usr/bin/ld: CMakeFiles/gfortran-regression-execute-regression__fmt_en_rd_f90.dir/fmt_en_rd.f90.o: in function `_QFPcheckfmt':
FIRModule:(.text+0x3757): undefined reference to `__truncsfbf2'
/usr/bin/ld: CMakeFiles/gfortran-regression-execute-regression__fmt_en_f90.dir/fmt_en.f90.o: in function `_QFPcheckfmt':
FIRModule:(.text+0x2797): undefined reference to `__truncsfbf2'
```

I am guessing I am likely not installing clang_rt/compiler-rt correctly, there is nothing special to do with regards to compiler-rt when building the test suite, right?

Reply:
> there is nothing special to do with regards to compiler-rt when building the test suite, right?

No, you shouldn't have to do anything special with compiler-rt. I don't build compiler-rt explicitly, so I am not sure why I don't see the same issue. I'll take a look at the binaries on my end too.

Reply:
I managed to get the same error when `--gcc-install-dir` was not passed and `flang` ended up picking the wrong GCC libraries. Not sure if that is the case with you, but might be worth looking into. 

Let me know how you'd like to proceed.

Reply:
Thanks for checking.
> I managed to get the same error when --gcc-install-dir was not passed

Passed to what? When building flang? I tried passing it to flang-new, and it indeed picks the intended gcc version I am using on that machine (9.3.0), but I still get the undefined symbol, so I assume it may not be part of the my gcc lib.

I greped in my 9.3.0 libgcc.a, and I have some hit, but not  `__truncsfbf2`:
````
0000000000000000 T __trunctfsf2
0000000000000000 T __trunctfdf2
0000000000000000 T __trunctfxf2
````

I was able to find  `__truncsfbf2` in libgcc.a 13 and 14, but not 9, 10, 11 and 12.

These symbols are also defined in the compiler-rt builtin library, hence I was wondering if that was how you were getting them. Using `-v` with flang-new you would see libclang_rt.builtins.a or libclang_rt.builtins-x86_64.a be linked , which happens when one use the `--rtlib=compiler-rt` flag with flang-new, but I was wondering if somehow your build of flang-new linked that by default, hence my initial question about compiler-rt. But it seems you are getting them in your gcc environment from what you are saying.

> Let me know how you'd like to proceed.

It appears this change will force the llvm-test-suite to be run with flang built with gcc 13 or higher, which is a significant limitation. I would advise for some cmake option to enable/disable tests requiring such float16 recent builtin symbols.

Reply:
> > I managed to get the same error when --gcc-install-dir was not passed
>>
> Passed to what? When building flang?

Ah, sorry, should have been clearer about this. I meant passing `--gcc-install-dir` to `flang-new`. I use GCC 14, which is why I was getting them. If I don't pass `--gcc-install-dir`, I get GCC 8.x, which is also consistent with what you are seeing. 

>  I would advise for some cmake option to enable/disable tests requiring such float16 recent builtin symbols.

I will disable these tests for now until I can figure out some way to selectively enable these depending on the availability of a sufficiently modern GCC or compiler-rt. Thanks for looking into this more closely.

Reply:
Disabled the following files

*  fmt_en.f90
*  fmt_en_rd.f90

These were the only ones that were failing, correct?

Reply:
Thanks for the update. I am hitting the issue on all fmt_en_... tests:
```
  fmt_en.f90
  fmt_en_rd.f90
  fmt_en_rn.f90
  fmt_en_ru.f90
  fmt_en_rz.f90
```

After that, everything passes for me when I apply #142. Thanks!

Reply:
>  I am hitting the issue on all fmt_en_... tests

Updated the list. 

> After that, everything passes for me when I apply https://github.com/llvm/llvm-test-suite/pull/142.

You anticipated my next request. Thanks! 🥇